File: leetcode/algorithms/682_baseball_game/main.java
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

class BaseballGame {
    public int calPoints(String[] operations) {
        List<Integer> scoreArray = new ArrayList<Integer>();

        for (String operation : operations) {
            if (operation.equals("+")) {
                scoreArray.add(scoreArray.get(scoreArray.size() - 2) + scoreArray.get(scoreArray.size() - 1));
            } else if (operation.equals("D")) {
                scoreArray.add(scoreArray.get(scoreArray.size() - 1) * 2);
            } else if (operation.equals("C")) {
                scoreArray.removeLast();
            } else {
                scoreArray.add(Integer.parseInt(operation));
            }
        }

        return scoreArray.stream().mapToInt(i -> i).sum();
    }


    // Solution
    public int solution(String[] ops) {
        Stack<Integer> stack = new Stack<Integer>();

        for (String op : ops) {
            if (op.equals("+")) {
                int top = stack.pop();
                int newtop = top + stack.peek();
                stack.push(top);
                stack.push(newtop);
            } else if (op.equals("C")) {
                stack.pop();
            } else if (op.equals("D")) {
                stack.push(2 * stack.peek());
            } else {
                stack.push(Integer.valueOf(op));
            }
        }

        int ans = 0;
        for (int score : stack) {
            ans += score;
        }

        return ans;
    }
}

File: leetcode/algorithms/682_baseball_game/main.py
from typing import List


class BaseballGame:
    def calPoints(self, operations: List[str]) -> int:
        scoreArray: List[int] = []

        for operation in operations:
            if operation == "+":
                scoreArray.append(scoreArray[len(scoreArray) - 2] + scoreArray[len(scoreArray) - 1])
            elif operation == "D":
                scoreArray.append(scoreArray[len(scoreArray) - 1] * 2)
            elif operation == "C":
                scoreArray.pop()
            else:
                scoreArray.append(int(operation))

        return sum(scoreArray)


    # Solution
    def solution(self, operations: List[str]) -> int:
        stack = []

        for operation in operations:
            if operation == '+':
                stack.append(stack[-1] + stack[-2])
            elif operation == 'C':
                stack.pop()
            elif operation == 'D':
                stack.append(2 * stack[-1])
            else:
                stack.append(int(operation))

        return sum(stack)
